src/mlpack/methods/reinforcement_learning/sac.hpp

```diff
@@ -94,14 +94,15 @@ class SAC
       EnvironmentType environment = EnvironmentType());
 
   /**
-   * Delete copy constructor.
-   */
-  SAC(SAC const&) = delete;
-
-  /**
-   * Enable move constructor by default.
+   * Copy the SAC object with given settings.
+   *
+   * If you want to pass in a parameter and discard the original parameter
+   * object, you can directly pass the parameter, as the constructor takes
+   * a reference. This avoids unnecessary copy.
+   *
+   * @param other SAC object to copy.
    */
-  SAC(SAC&&) = default;
+  SAC(const SAC& other);
```

Comment:
Oops, sorry, this one fell off my list.  If we specify a copy constructor, then the default copy operator and move constructor/operators won't be defined.  (Correct me if I'm wrong on that.)  So should we explicitly specify, e.g., `SAC(SAC&&) = default` and the same for the copy and move operators?

Reply:
You are right, we only need a copy constructor, there will be no need to specify the move constructor.

Reply:
Ok---if only the copy constructor is special, then we should add `SAC(SAC&&) = default` and then simple implementations for `operator=(const SAC&)` and `operator=(SAC&&)` (I don't know if you can say `= default` for those; maybe it's worth a shot). :+1: